test/functional/lua/ssh_spec.lua

```diff
@@ -0,0 +1,44 @@
+local t = require('test.testutil')
+local parser = require('vim._ssh.parser')
```

Comment:
`runtime/lua/vim/net/_ssh.lua` seems like a good place for this, though it's private for now. 

do not create lots of granular modules. just put everything in `_ssh.lua` , until it's clear that separate modules are needed.